### PR TITLE
Drop support for NET451

### PR DESCRIPTION
**Pull request by azabbasi**
In this PR you will find:
* Cproj files updated to remove 451 specific items
* Updated vsts tasks to drop 451 from the matrix
* Removed all pre-compile directives for 451 

This PR will be followed by a code cleanup task to remove all dead code that we might have missed.

### CODE REVIEW

**Review by timtay-microsoft — APPROVED** [review]
Beautiful

File: e2e/test/E2ETests.csproj [hunk 1]
```diff
@@ -56,7 +35,7 @@
     <ProjectReference Include="$(RootDir)\iothub\service\src\Microsoft.Azure.Devices.csproj" />
     <ProjectReference Include="$(RootDir)\shared\src\Microsoft.Azure.Devices.Shared.csproj" />
   </ItemGroup>
-  <ItemGroup Condition=" ('$(AZURE_IOT_LOCALPACKAGES)' == '') And ( '$(TargetFramework)' != 'net451' ) ">
+  <ItemGroup Condition=" '$(AZURE_IOT_LOCALPACKAGES)' == '' ">
```
Comment (abhipsaMisra) [inline e2e/test/E2ETests.csproj:38]: this condition isn't require anymore. can be merged with the one above

File: e2e/test/E2ETests.csproj [hunk 2]
```diff
@@ -73,7 +52,7 @@
     <PackageReference Include="Microsoft.Azure.Devices.Shared" Version="1.*" />
     <PackageReference Include="Microsoft.Azure.Devices.Client" Version="1.*" />
   </ItemGroup>
-  <ItemGroup Condition=" ('$(AZURE_IOT_LOCALPACKAGES.ToUpper())' != '') And ( '$(TargetFramework)' != 'net451' ) ">
+  <ItemGroup Condition=" '$(AZURE_IOT_LOCALPACKAGES.ToUpper())' != '' ">
```
Comment (abhipsaMisra) [inline e2e/test/E2ETests.csproj:55]: this too can be merged with the one above

**Review by abhipsaMisra — APPROVED** [review]
🎉

**Pull request merged**